File: CPP_programming/Day10_Template/A3_Cstack/Cstack.cpp
#include"Cstack.h"


template <class T>
Cstack<T>::Cstack()
{
    size=5;
    top=-1;
    arr= new T[size];
}

template <class T>
Cstack<T>::Cstack(int s)
{
    size=s;
    top=-1;
    arr= new T[size];
}

template<class T>
void Cstack<T>::push(T n)
{
    top++;
    arr[top]=n;
}

template<class T>
T Cstack<T>::pop()
{
    T temp= arr[top];
    top--;
    return temp;
}

template <class T>
void Cstack<T>::display()
{
    for(int i=top; i>=0; i--)
    {
        cout<<arr[i]<<"\n";
    }
}

File: Data Structure using java/_3_stack/src/a4_paranthesis/cStack.java
package a4_paranthesis;


public class cStack 
{
	private int top;
	private int size;
	private char arr[];
	
	public cStack() 
	{
		top=-1;
		size=3;
		arr = new char[3];
	}

	public cStack(int size) 
	{
		top=-1;
		this.size = size;
		arr = new char[size];
	}
	
	public boolean isEmpty()
	{		
		if(top==-1)
		{
			return true;
		}
		return false;
	}
	
	public boolean isFull()
	{
		if(top==size-1)
		{
			return true;
		}
		 return false;
	}
	
	public void push(char data) 
	{
		if(!isFull())
		{
			arr[++top]=data;
		}
		else
			System.out.println("Stack is Full Paji!!!");
	}
	
	public char pop() 
	{
		char data = '\n';
		if(!isEmpty())
		{
			data = arr[top];
			top--;
			return data;
		}
		else
//			System.out.println("\t\tStack is Empty Paji!!!");
		return data;
	}
	
	public char peek() 
	{
		char data = '\n';
		if(!isEmpty())
		{
			return arr[top];
		}
		else
//			System.out.println("\t\tStack is Empty Paji!!!");
		return data;
	}
	
	public void display()
	{
		System.out.println("-------------------Stack-------------------");
//		for(int i=this.top; i>=0; i--)
		int i=top;
		while(i>=0)
		{
//			System.err.println("|"+"--"+"|");
			System.err.println("|"+arr[i]+"|");
			i--;
		}
//		System.err.println("|"+"--"+"|");
	}
	
	
	
	
	
	
	

}
